pubtools/_pulp/tasks/publish.py

```diff
@@ -0,0 +1,221 @@
+import logging
+import os
+import sys
+import re
+from datetime import datetime
+from functools import partial
+from more_executors.futures import f_map, f_sequence
+
+from pubtools.pulplib import Criteria, Matcher, PublishOptions
+
+from pubtools._pulp.task import PulpTask
+from pubtools._pulp.services import (
+    PulpClientService,
+    FastPurgeClientService,
+    UdCacheClientService,
+)
+
+
+step = PulpTask.step
+
+LOG = logging.getLogger("pubtools.pulp")
+
+
+def publish_date(str_date):
+    # validates publish-before date
+    return datetime.strptime(str_date, "%Y-%m-%d")
+
+
+class Publish(
+    PulpClientService, FastPurgeClientService, UdCacheClientService, PulpTask
+):
+    """Publishes the pulp repositories to the endpoints defined by the distributors
+
+    This command will publish the pulp repositories provided in the request or
+    fetched using the filters(url-regex or published-before) or an intersection
+    of input repos and filters.
+    """
+
+    def add_args(self):
+        super(Publish, self).add_args()
+
+        self.parser.add_argument(
+            "--repos", help="list of repos to be published", nargs="+"
+        )
+        self.parser.add_argument(
+            "--clean",
+            help="attempt to remove content not in the repo",
+            type=bool,
+            default=None,
```

Comment:
I don't think clean and force work as you wanted now.  With this definition, you have to pass an argument for them:

```
  --clean CLEAN         attempt to remove content not in the repo
  --force FORCE         execute a full repo publish
```

Yet because ``bool(any_nonempty_string)`` is always True, it doesn't matter what value you pass.  For example ``--force false`` **enables** force.

Try putting ``action='store_true'`` instead, then the behavior is:

- default is False/None
- if you pass ``--clean`` (with no argument), it becomes ``True``

Reply:
+1